spring-cloud-gcp-docs/src/main/asciidoc/spanner.adoc

```diff
@@ -320,9 +320,95 @@ Note that embedded properties' column names must all be unique.
-Currently there is no support to map relationships between objects.
-I.e., currently we do not have ways to establish parent-children relationships directly via annotations.
-This feature is actively being worked on.
+Spring Data Cloud Spanner supports parent-child relationships using the Cloud Spanner https://cloud.google.com/spanner/docs/schema-and-data-model#creating-interleaved-tables[parent-child interleaved
+table mechanism]. Cloud Spanner interleaved tables enforce the one-to-many relationship and provide efficient queries and operations
+on entities of a single domain parent entity. These relationships can be up to 7 levels deep. Cloud Spanner also provides
+automatic cascading delete or enforces the deletion of child entities before parents.
+
+While one-to-one and many-to-many relationships can be implemented in Cloud Spanner and Spring Data Cloud Spanner using constructs of
+interleaved parent-child tables, only the parent-child relationship is natively supported. Cloud Spanner does not
+support the foreign key constraint, though the parent-child key constraint enforces a similar requirement when used
+with interleaved tables.
+
+For example, the following Java entities:
+
+[source,java]
+----
+@Table(name = "Singers")
+class Singer {
+  @PrimaryKey
+  long SingerId;
+
+  String FirstName;
+
+  String LastName;
+
+  byte[] SingerInfo;
+
+  @Interleaved
+  List<Album> albums;
+}
+
+@Table(name = "Albums")
+class Album {
+  @PrimaryKey
+  long SingerId;
+
+  @PrimaryKey(keyOrder = 2)
+  long AlbumId;
+
+  String AlbumTitle;
+}
+----
+
+These classes can correspond to an existing pair of interleaved tables. The `@Interleaved` annotation may be applied to
+`Collection` properties and the inner type is resolved as the child entity type. The schema needed to create them can
+also be generated using the `SpannerSchemaUtils` and executed using the `SpannerDatabaseAdminTemplate`:
+
+[source,java]
+----
+
+@Autowired
+SpannerSchemaUtils schemaUtils;
+
+@Autowired
+SpannerDatabaseAdminTemplate databaseAdmin;
+...
+
+// Get the create statmenets for all tables in the table structure rooted at Singer
+List<String> createStrings = this.schemaUtils.getCreateTableDdlStringsForInterleavedHierarchy(Singer.class);
+
+// Create the tables and also create the database if necessary
+this.databaseAdmin.executeDdlStrings(createStrings, true);
+----
+
+The `createStrings` list contains table schema statements using column names and types compatible with the provided
+Java type and any resolved child relationship types contained within based on the configured custom converters.
+
+[source,sql]
+----
+CREATE TABLE Singers (
+  SingerId   INT64 NOT NULL,
+  FirstName  STRING(1024),
+  LastName   STRING(1024),
+  SingerInfo BYTES(MAX),
+) PRIMARY KEY (SingerId);
+
+CREATE TABLE Albums (
+  SingerId     INT64 NOT NULL,
+  AlbumId      INT64 NOT NULL,
+  AlbumTitle   STRING(MAX),
+) PRIMARY KEY (SingerId, AlbumId),
+  INTERLEAVE IN PARENT Singers ON DELETE CASCADE;
```

Comment:
You probably want to document the property that controls this "DELETE ON CASCADE".